File: algorithm_workbook/python_SWEA/1486.py
def make_top(depth, acc):
    global res
    if depth == N:
        if acc >= B:
            res = min(res,acc-B)
        return
    make_top(depth+1, acc + arr[depth])
    make_top(depth+1, acc)


T = int(input())
for tc in range(1,T+1):
    N, B = map(int, input().split())
    arr = list(map(int,input().split()))
    candidate = []
    res = 987654321
    make_top(0,0)
    print(f'#{tc} {res}')

File: algorithm_workbook/cpp_SWEA/1486.cpp
#include <iostream>
#include <algorithm>
#include <string.h>
#define INF 987654321

using namespace std;

int N, B;
int arr[20];
int res;

void mk_top(int depth, int acc){
  if(depth == N){
    if(acc >= B){
      res = min(res, acc);
    }
    return;
  }
  mk_top(depth+1, acc + arr[depth]);
  mk_top(depth+1, acc);
}


int main(){
  int tc;
  int T;

  cin >> T;

  for(tc=1; tc<=T; tc++){
    N = 0, B = 0;
    memset(arr, 0, sizeof(arr));
    res = INF;

    cin >> N >> B;
    for(int i = 0; i < N; i++){
      cin >> arr[i];
    }

    mk_top(0,0);

    cout << "#" << tc << " " << res - B << "\n";
  }
  return 0;
}
